src/helpers/offerings-parser.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  type OfferingResponse,
+  type OfferingsResponse,
+  type PlacementsResponse,
+} from "../networking/responses/offerings-response";
+import {
+  type ProductResponse,
+  type ProductsResponse,
+} from "../networking/responses/products-response";
+import {
+  type Offering,
+  type Offerings,
+  type Package,
+  toOffering,
+} from "../entities/offerings";
+import { Logger } from "./logger";
+
+const addPlacementContextToPackage = (
+  rcPackage: Package,
+  placementId: string,
+): Package => {
+  return {
+    ...rcPackage,
+    rcBillingProduct: {
+      ...rcPackage.rcBillingProduct,
+      presentedOfferingContext: {
+        ...rcPackage.rcBillingProduct.presentedOfferingContext,
```

Comment:
Can the `rcPackage.rcBillingProduct.presentedOfferingContext` be null/undefined?

Reply:
Good question! And no, since currently, all products can only be obtained through an offering, we at least fill the `presentedOfferingContext` with the presented offering identifier (the other fields may be null if not using targeting/placements. Thanks for double checking!